docs/core/deploying/native-aot/libraries/ios-like-native-libraries-with-nativeaot.md

```diff
@@ -0,0 +1,247 @@
+        install_name_tool -rpath @executable_path @executable_path/../Frameworks MyNativeAOTLibrary/bin/Release/net9.0/maccatalyst-x64/publish/MyNativeAOTLibrary.dylib
+        ```
+
+    - `LC_ID_DYLIB` load command
+
+        ```bash
+        install_name_tool -id @rpath/MyNativeAOTLibrary.framework/Versions/A/MyNativeAOTLibrary MyNativeAOTLibrary/bin/Release/net9.0/maccatalyst-arm64/publish/MyNativeAOTLibrary.dylib
+        install_name_tool -id @rpath/MyNativeAOTLibrary.framework/Versions/A/MyNativeAOTLibrary MyNativeAOTLibrary/bin/Release/net9.0/maccatalyst-x64/publish/MyNativeAOTLibrary.dylib
+        ```
+
+3. Manually package the binary into a universal file
+
+    ```bash
+    lipo -create MyNativeAOTLibrary/bin/Release/net9.0/maccatalyst-arm64/publish/MyNativeAOTLibrary.dylib MyNativeAOTLibrary/bin/Release/net9.0/maccatalyst-x64/publish/MyNativeAOTLibrary.dylib -output MyNativeAOTLibrary.framework/Versions/A/MyNativeAOTLibrary
+    ```
+
+4. Add a property list file to your framework:
+
+    - Create a `Info.plist` file
+
+    ```bash
+    touch MyNativeAOTLibrary.framework/Versions/A/Resources/Info.plist
+    ```
+
+    - Add the contents from the [appendix](#appendix-infoplist-contents) into the created `Info.plist` file
+
+After the final step the framework structure should look like this:
+
+```
+MyNativeAOTLibrary.framework
+    |_ MyNativeAOTLibrary -> Versions/Current/MyNativeAOTLibrary
+    |_ Resources -> Versions/Current/Resources
+    |_ Versions
+        |_ A
+        |   |_ Resources
+        |   |   |_ Info.plist
+        |   |_ MyNativeAOTLibrary
+        |_ Current -> A
+```
+
+### Consuming custom frameworks
+
+1. Open `Xcode` (in this example `Xcode 16.0` is used)
+2. Create a new `App` project
+3. Choose the name for your app (e.g., `MyiOSApp`) and choose Objective-C as the source language
+4. Add a reference to the `MyNativeAOTLibrary` framework
+    - In the `MyiOSApp` targets `General` tab, under `Frameworks, Libraries and Embedded Content` click the `+` sign to add `MyNativeAOTLibrary` as the referenced framework
+    - In the dialog choose: `Add Other` -> `Add Files` and then browse to the location of `MyNativeAOTLibrary.framework` and select it
+    - Once selected, set `Embed and Sign` option for `MyNativeAOTLibrary` framework
+
+    ![Drop](./xcode-add-framework-reference.png)
+5. Add `MyNativeAOTLibrary.framework` location to the list of `Framework Search Paths` in the `Build Settings` tab
+
+    ![Drop](./xcode-add-framework-search-path.png)
+6. Edit the `main.m` by calling the exposed managed method `aotsample_add` and printing the result
+
+    ```objc
+        extern int aotsample_add(int a, int b);
+        int main(int argc, char * argv[]) {
+            ...
+            NSLog(@"2 + 5 = %d", aotsample_add(2, 5));
+            ...
+        }
```

Comment:
Looks like this is unnecessarily indented.

Reply:
Thanks, fixed